packages/abi-coder/src/encoding/coders/v1/VecCoder.ts

```diff
@@ -34,7 +37,7 @@ export class VecCoder<TCoder extends Coder> extends Coder<
   }
 
   decode(data: Uint8Array, offset: number): [DecodedValueOf<TCoder>, number] {
-    if (data.length < this.encodedLength || data.length > MAX_BYTES) {
+    if (!this.#isOptionVec && (data.length < this.encodedLength || data.length > MAX_BYTES)) {
```

Comment:
Options now have no data bytes when the value is `None`. Meaning when `None`, the options expected length will not be equal to the data length passed.

This is validated correctly on the underlying data structure (eg `boolean option` will be validated at the `boolean` level) so removed the validation at the `vec` level.